### PR TITLE
Remove extension methods

### LINKED ISSUE #111: develop branch: ParserExtensions methods should not be extensions
I don't think we should have extension methods called `Parse` and `Lex` on string.

Why should `"test".Parse()` return a `GraphQLDocument`?  Why shouldn't it parse to an integer, for example?  And why should `"test".Lex()` create a lexer for GraphQL documents?  Why not a lexer for json?

When you parse an integer, you call `int.Parse("234")`.  It's not just naming -- Microsoft didn't add a "ParseInt" extension method for strings.

I believe we should remove `ParseExtensions`. The static methods on `Parser` and `Lexer` are well designed.  No extension methods are needed.  If desired for tests, someone can add an extension method within their a test project to facilitate testing.  It should not be part of the library.

### PR DESCRIPTION
**Pull request by Shane32**
Fixes #111

### CODE REVIEW

File: src/GraphQLParser.Tests/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using GraphQLParser.AST;
+
+namespace GraphQLParser.Tests
+{
+    internal static class ParserTestExtensions
+    {
+
```
Comment (sungam3r) [inline src/GraphQLParser.Tests/Extensions.cs:10]:
```suggestion
```

File: src/GraphQLParser.Tests/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        public static Token Lex(this string source, int start = 0) => Lexer.Lex(source, start);
+
+        /// <summary>
+        /// Generates AST based on input text.
+        /// </summary>
+        /// <param name="source">Input data as a string.</param>
+        /// <param name="options">Parser options.</param>
+        /// <returns>AST (Abstract Syntax Tree) for GraphQL document.</returns>
+        public static GraphQLDocument Parse(this string source, ParserOptions options = default) => Parser.Parse(source, options);
+
```
Comment (sungam3r) [inline src/GraphQLParser.Tests/Extensions.cs:26]:
```suggestion
```

**Comment by sungam3r** [comment]
OK. Please change also readme.

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
Now see if #111 closes automatically.

**Comment by sungam3r** [comment]
Nope

**Comment by Shane32** [comment]
Yeah, weird.

**Comment by sungam3r** [comment]
https://github.blog/2013-01-22-closing-issues-via-commit-messages/

**Pull request merged**